service/backend/pchain/backend.go

```diff
@@ -104,3 +107,24 @@ func (b *Backend) getGenesisBlock(ctx context.Context) (*indexer.ParsedGenesisBl
 	}
 	return genesisBlock, nil
 }
+
+func (b *Backend) initChainIDs() error {
```

Comment:
moved getChainIDs to backend constructor

Reply:
[Rosetta has the concept of an offline mode](https://www.rosetta-api.org/docs/construction_api_introduction.html#completely-offline), where all construction endpoints except `/construction/metadata` must be able to run without connection to the node. Because of this, we cannot put client calls in the backend constructor by default. We lazy loaded the mappings with caching for this reason.

Reply:
Thanks for clarifying this. I undestand node mode cannot change at run time. Hence I fixed this by passing service mode to Backend constructor.

Reply:
Thanks for the explainantion. Since service mode cannot change at run time, we can still build chainsID mapping at backend construction, just passing in service mode.